modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/skin/WebColorFieldSkin.java

```diff
@@ -36,8 +37,11 @@
 /**
  */
 public class WebColorFieldSkin extends InputFieldSkin {
+    private static final String HEX_DIGIT = "[A-Fa-f0-9]";
+    private static final Pattern PATTERN = Pattern.compile("#?" + HEX_DIGIT + "{6}");
+    private static final Pattern PARTIAL_PATTERN = Pattern.compile("#?" + HEX_DIGIT + "{0,6}");
```

Comment:
You might also use the character class `\p{XDigit}`, which matches a hexadecimal character.